library/ssl_tls.c

```diff
@@ -7639,20 +7639,22 @@ int mbedtls_ssl_parse_certificate(mbedtls_ssl_context *ssl)
 }
 #endif /* MBEDTLS_KEY_EXCHANGE_WITH_CERT_ENABLED */
 
-#if defined(MBEDTLS_MD_CAN_SHA256)
-static int ssl_calc_finished_tls_sha256(
-    mbedtls_ssl_context *ssl, unsigned char *buf, int from)
+static int ssl_calc_finished_tls_generic(mbedtls_ssl_context *ssl, void *ctx,
```

Comment:
Actually, this is going to be always true, since `MBEDTLS_SSL_TLS_C` already depends those hashes being present: https://github.com/Mbed-TLS/mbedtls/blob/development/include/mbedtls/check_config.h#L883. I'll add a comment about this because it's indeed not obvious at all.